lib/libc/stdlib/malloc/mrs/mrs.c

```diff
@@ -1255,11 +1255,6 @@ static void *mrs_realloc(void *ptr, size_t size) {
 	size_t old_size = cheri_getlen(ptr);
 	mrs_debug_printf("mrs_realloc: called ptr %p ptr size %zu new size %zu\n", ptr, old_size, size);
 
-	if (size == 0) {
```

Comment:
I mentioned this over in #1888.  I haven't yet tested the fixup version, but I actually think removing this special case is the cleanest solution given what return values of `NULL` from `realloc` mean.